src/platform/packages/shared/response-ops/alerts-delete/api/last_run/use_alert_delete_last_run.ts

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the "Elastic License
+ * 2.0", the "GNU Affero General Public License v3.0 only", and the "Server Side
+ * Public License v 1"; you may not use this file except in compliance with, at
+ * your election, the "Elastic License 2.0", the "GNU Affero General Public
+ * License v3.0 only", or the "Server Side Public License, v 1".
+ */
+
+import { useQuery } from '@tanstack/react-query';
+import type { HttpStart } from '@kbn/core-http-browser';
+import { useRef } from 'react';
+import { getAlertDeleteLastRun } from './get_alert_delete_last_run';
+
+export interface UseAlertDeleteLastRunParams {
+  isEnabled: boolean;
+  isOpen: boolean;
+  services: { http: HttpStart };
+}
+export const useAlertDeleteLastRun = ({
+  isEnabled,
+  isOpen,
+  services: { http },
+}: UseAlertDeleteLastRunParams) => {
+  // This ref is used to track every time the modal is opened.
+  // We use it as query key + Infinity staleTime to only fetch
+  // once every time the modal is opened.
+  const modalOpenCountRef = useRef(0);
```

Comment:
I am ok with limiting the number of times we fetch, but this approach with `staleTime: Infinity` will force the call to be made **every single time** we open the modal.

Is this deliberate? Do the runs happen often enough for this to be justified, or could we do something simpler with a shorter `staleTime` and only query sometimes when the modal opens?

Reply:
It's on purpose yes. I don't think users will be closing/opening as ofter and fast to make it a difference. The idea was to try to make sure that every time they open the modal, they will get the latest update because if we do stale for, let's say 2 minutes, and users come back a minute after starting the schedule and see no update it will look like the schedule didn't run at all. It also doesn't need to fetch while updating anything in the form.